src/main/java/frc/robot/classes/spikemotor/SpikeMotorTalonFX.java

```diff
@@ -0,0 +1,90 @@
+package frc.robot.classes.spikemotor;
+
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.InvertType;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.TalonFXControlMode;
+import com.ctre.phoenix.motorcontrol.TalonFXFeedbackDevice;
+
+import frc.robot.classes.SPIKE293Utils;
+import static frc.robot.Constants.DrivetrainConstants.*;
+
+public class SpikeMotorTalonFX extends SpikeMotor {
+    private TalonFX motor;
+    private double wheelDiameter;
+    private Boolean inverted;
+    private InvertType invertType;
+
+    public SpikeMotorTalonFX(double wheelDiameter, boolean inverted) {
+        this.wheelDiameter = wheelDiameter;
+        this.inverted = inverted;
+    }
+
+    public SpikeMotorTalonFX(double wheelDiameter, InvertType invertType) {
+        this.wheelDiameter = wheelDiameter;
+        this.invertType = invertType;
+    }
+
+    @Override
+    protected void initImpl(int deviceNumber) {
+        motor = new TalonFX(deviceNumber);
+        motor.clearStickyFaults();
+        motor.configFactoryDefault();
+        motor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0,
+            CONFIG_FEEDBACKSENSOR_TIMEOUT_MS);
+        motor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 1,
+            CONFIG_FEEDBACKSENSOR_TIMEOUT_MS);
+        if (inverted != null) {
+            motor.setInverted(inverted);
+            motor.setSensorPhase(inverted);
+        } else {
+            motor.setInverted(invertType);
+        }
+        motor.setNeutralMode(NeutralMode.Coast);
+        if (invertType != InvertType.FollowMaster) {
+            motor.config_kF(VELOCITY_PID_SLOT_ID, VELOCITY_KF, PID_CONFIG_TIMEOUT_MS);
```

Comment:
We may want to make these values settable in a method since more than one falcon motor will be used with different PID values

Reply:
Can do!

Reply:
Wait hold on -- don't we need a consistent API for these? If so, then shouldn't all motors also have custom PID values?